### PR TITLE
Run all tests and summarize failures, add new fuzzers to artifact

### LINKED ISSUE #6623: [ossfuzz] nightly CI fails on first error instead of running through and summarizing all errors at the end of the run
## Description

Bug 1: At the moment, the nightly regression tests fail on first failure. Instead we should continue past failing tests and show a summary of all failures.

Proposed solution: Maintain a python array of state of each test and return `all()` of the boolean array.

Bug 2: Test status is not propagated to python script's exit status e.g., via `sys.exit()`

Proposed solution: Modify the return code of the test runner from True/False to `bash` exit codes. A zero return means success, failure otherwise. All tests pass when `any(returnCodeList)` is also zero i.e., no test fails.

### PR DESCRIPTION
**Pull request by bshastry**
closes #6623 but introduces #6624

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -47,6 +47,8 @@ defaults:
         - test/tools/ossfuzz/solc_noopt_ossfuzz
         - test/tools/ossfuzz/const_opt_ossfuzz
         - test/tools/ossfuzz/strictasm_diff_ossfuzz
+        - test/tools/ossfuzz/strictasm_opt_ossfuzz
+        - test/tools/ossfuzz/strictasm_assembly_ossfuzz
```
Comment (erak) [inline .circleci/config.yml]: Could those be sorted alphabetically?

**Review by erak — DISMISSED** [review]
Looks good (besides the tiny comment).

File: scripts/regressions.py [hunk 2]
```diff
@@ -48,6 +48,13 @@ def parseCmdLine(self, description, args):
                        help="""Directory where test results will be written""")
         return argParser.parse_args(args)
 
+    """
```
Comment (chriseth) [inline scripts/regressions.py]: Shouldn't a python docstring be inside the function?

**Comment by bshastry** [comment]
Updated PR
  - artifacts sorted alphabetically
  - docstring moved to inside function

**Review by chriseth — APPROVED** [review]

**Pull request merged**